packages/core/src/components/fields/SchemaField.js

```diff
@@ -241,7 +241,9 @@ function SchemaFieldRender(props) {
   const FieldTemplate =
     uiSchema["ui:FieldTemplate"] || registry.FieldTemplate || DefaultTemplate;
   let idSchema = props.idSchema;
-  const schema = retrieveSchema(props.schema, definitions, formData);
+  const schema = handleArrayType(
```

Comment:
Is there a reason why you only call `handleArrayType` over here? Perhaps it would be better to call it within `retrieveSchema` instead?